main.go

```diff
@@ -88,108 +96,175 @@ func (c *DownloadPlugin) Run(cliConnection plugin.CliConnection, args []string)
 		os.Exit(1)
 	}
 
-	flagVals := ParseFlags(args)
+	cmdExec := cmd_exec.NewCmdExec()
+
+	// parse flags and get list of download paths
+	flagVals, paths := ParseArgs(args)
+	paths = ExpandGlobs(cmdExec, paths, flagVals.Instance_flag)
 
-	// flag variables
-	filterList := filter.GetFilterList(flagVals.Omit_flag, flagVals.Verbose_flag) // get list of things to not download
+	// get list of things to not download
+	filterList := filter.GetFilterList(flagVals.Omit_flag, flagVals.Verbose_flag)
 
+	// get server path to download from and local path to download to
 	workingDir, err := os.Getwd()
 	check(err, "Called by: Getwd")
-	rootWorkingDirectoryServer, startingPathServer := GetDirectoryContext(workingDir, args)
-	rootWorkingDirectoryLocal := filepath.FromSlash(rootWorkingDirectoryServer)
+	pathVals := GetDirectoryContext(workingDir, paths, flagVals.File_flag)
+	for _, v := range pathVals {
+		v.RootWorkingDirectoryLocal = filepath.FromSlash(v.RootWorkingDirectoryLocal)
```

Comment:
The rootWorkingDirectoryServer and rootWorkingDirectoryLocal variables were created to try to prevent confusion. As @whereswaldon mentioned, the local variable needed to be OS-agnostic while the server variable was for a Linux server. @reSoley is correct though when he says it is merely passed into another function and makes no discernible difference.